scripts/prepare-package-for-publish.mjs

```diff
@@ -1,32 +1,115 @@
-  'dist/handsontable.full.css',
-  'dist/handsontable.full.js',
-  'dist/handsontable.full.min.css',
-  'dist/handsontable.full.min.js',
-  'dist/handsontable.js',
-  'dist/handsontable.min.css',
-  'dist/handsontable.min.js',
-  'dist/languages',
-  'dist/README.md',
-  'languages',
-  'base.d.ts',
-  'CHANGELOG.md',
-  'handsontable-general-terms.pdf',
-  'handsontable-non-commercial-license.pdf',
-  'handsontable.d.ts',
-  'LICENSE.txt',
-  'package.json',
-  'README.md',
-];
-
-filesToMove.forEach((file) => {
+const TARGET_PATH = './tmp/';
+const PACKAGE_PATH = path.resolve('package.json');
+const DEV_PACKAGE = fse.readJsonSync(PACKAGE_PATH, { encoding: 'utf-8' });
+const { handsontable } = DEV_PACKAGE;
+const {
+  copy: FILES_TO_COPY,
+  exports: EXPORTS_RULES,
+  fields: PACKAGE_FIELDS_TO_COPY,
+} = handsontable;
+
+/**
+ * Copy necessary files we don't need to process.
+ */
+FILES_TO_COPY.forEach((file) => {
   fse.copySync(
     path.resolve(`./${file}`),
-    path.resolve(`${TARGET_PATH}/${file}`),
+    path.resolve(`${TARGET_PATH}${file}`),
     { overwrite: true });
 });
+
+/**
+ * Prepare exports basing on wildcards in paths.
+ */
+const groupedExports = EXPORTS_RULES.flatMap((rule) => {
+  if (typeof rule !== 'string') {
+    return rule;
+  }
+
+  if (!rule.includes('*')) {
+    return { [rule]: rule };
+  }
+
+  const rules = {};
+  const foundFiles = glob.sync(`${rule}`, { cwd: TARGET_PATH });
+
+  foundFiles.forEach((file) => {
+    const cleanPath = file.replace(/\.(m|)js$/, '').replace('/index', '');
+
+    if (!rules[cleanPath]) {
+      rules[cleanPath] = {};
+    }
+
+    const key = file.includes('.mjs') ? 'import' : 'require';
+
+    rules[cleanPath][key] = file;
+  });
+
+  return rules;
+});
+const targetExports = Object.assign({}, ...groupedExports);
+
+/**
+ * Test exports to verify if paths exist in the target directory.
+ */
+const EXPORTS_ERRORS = [];
+
+Object.keys(targetExports).forEach((ruleName) => {
+  const rule = targetExports[ruleName];
+
+  if (typeof rule === 'string') {
+    const pathToFile = `${TARGET_PATH}/${rule}`;
+    const fileExists = fse.existsSync(pathToFile);
+
+    if (!fileExists) {
+      EXPORTS_ERRORS.push(`${ruleName}: ${pathToFile}`);
+    }
+  } else {
+    if (rule.import) {
+      const pathToImport = `${TARGET_PATH}/${rule.import}`;
+      const importFileExists = fse.existsSync(pathToImport);
+
+      if (!importFileExists) {
+        EXPORTS_ERRORS.push(`${ruleName}: ${pathToImport}`);
+      }
+    }
+    if (rule.require) {
+      const pathToRequire = `${TARGET_PATH}/${rule.require}`;
+      const requireFileExists = fse.existsSync(pathToRequire);
+
+      if (!requireFileExists) {
+        EXPORTS_ERRORS.push(`${ruleName}: ${pathToRequire}`);
+      }
+    }
+  }
+
+});
+
+if (EXPORTS_ERRORS.length > 0) {
+  const FILES_LIST = `${EXPORTS_ERRORS.map(msg => `- ${msg}`).join('\n')}`;
+
+  throw new Error(`The following exports point to the non-existing files:\n ${FILES_LIST}`);
```

Comment:
To be consistent with other scripts and how they are managing the error messages can I suggest using the `displayErrorMessage` helper? 

After helper importing it might look like this:
```suggestion
  displayErrorMessage(`The following exports point to the non-existing files:\n ${FILES_LIST}`);
  process.exit(1);
```

Additionally, it would have been nice if the file paths in the error message were better formatted. This's how it looks like right now:
![image](https://user-images.githubusercontent.com/571316/116386266-ea17a900-a819-11eb-9922-042fc3b17c10.png)